bin/tictactoe

```diff
@@ -1,3 +1,45 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+def start
+  puts "Welcome to Tic-Tac-Toe!"
+  puts "How many players? (0, 1, or 2)"
+
+  input = gets.strip
+
+  case input
+    when "0"
+      Game.new(Players::Computer.new("X"), Players::Computer.new("O")).play
+      puts "Would you like to play again?"
+      new_game
+    when "1"
+      Game.new(Players::Human.new("X"), Players::Computer.new("O")).play
+      puts "Would you like to play  again?"
+      new_game
+    when "2"
+      Game.new(Players::Human.new("X"), Players::Human.new("O")).play
+      puts "Would you like to play again?"
+      new_game
+    else
+      puts "That is an invalid entry"
+      start
+  end
+end
+
+def new_game
+  puts "Enter Y for  yes and N for no"
+  input = gets.strip.downcase
+
+  case input
+  when 'y'
+    start
+  when 'n'
+    puts "Thanks for playing! Goodbye!"
+  else
+    new_game
+  end
+end
+
+
+start
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.